lib/Arch/X86/Semantics/DATAXFER.cpp

```diff
@@ -1121,3 +1121,51 @@ IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMd_64) = MOVSX<R64W, M32, int64_t>;)
 IF_64BIT(DEF_ISEL(MOVSXD_GPRv_MEMz_64) = MOVSX<R64W, M32, int64_t>;)
 IF_64BIT(DEF_ISEL(MOVSXD_GPRv_GPR32_64) = MOVSX<R64W, R32, int64_t>;)
 IF_64BIT(DEF_ISEL(MOVSXD_GPRv_GPRz_64) = MOVSX<R64W, R32, int64_t>;)
+
+namespace {
+template <typename D, typename K, typename S>
+DEF_SEM(VPMOVSXBQ_MASKmskw_SIMD128, D dst, K k1, S src) {
+  auto src_vec = SReadV8(src);
+  auto dst_vec = SClearV64(SReadV64(dst));
+  auto k_vec = Read(k1);
+  for (int i = 0; i< 2; i++) {
+    if (READBIT(k_vec, i) == 0) {
+      dst_vec = SInsertV64(dst_vec, i, 0);
+    }else {
+      auto v = SExtTo<int64_t>(SExtractV8(src_vec, i));
+      dst_vec = SInsertV64(dst_vec, i, v);
+    }
+  }
+  SWriteV64(dst, dst_vec);
+  return memory;
+}
+}
+
+#if HAS_FEATURE_AVX512
+DEF_ISEL(VPMOVSXBQ_XMMi64_MASKmskw_MEMi8_AVX512) = VPMOVSXBQ_MASKmskw_SIMD128<VV128W, R8, MV16>;
+DEF_ISEL(VPMOVSXBQ_XMMi64_MASKmskw_XMMi8_AVX512) = VPMOVSXBQ_MASKmskw_SIMD128<VV128W, R8, V128>;
+#endif  // HAS_FEATURE_AVX512
+
+namespace {
+template <typename D, typename K, typename S>
+DEF_SEM(VPMOVSXWD_MASKmskw_SIMD128, D dst, K k1, S src) {
```

Comment:
Probably, you could go and group these avx512-related semantics together, then follow them up with the isel defs, all within the `#if` blocks.

Reply:
Yes, modified according to your opinion